contrib/syslog-debun

```diff
@@ -456,6 +546,172 @@ debun_extra_debian () {
+		#this info should only be collected if the user has not requested privacy mode
+		#we value our customer's sense of privacy
+		if [ -z "$privacy_mode" ]; then
+			netstat -punt >${tmpdir}/sys.netstat.est.noss
+		fi
+	fi
+
+}
+
+debun_extra_genlinux () {
+	if is_available getenforce; then
+		getenforce >${tmpdir}/sys.selinux
+	else
+		echo "No getenforce in path." >${tmpdir}/sys.selinux
+	fi
+	sysctl -a >${tmpdir}/sys.sysctl.all 2>/dev/null
+}
+
+debun_linux () {
+	if [ "$dist" = "Debian" ]; then
+		add_extra debun_extra_debian
+	elif [ "$dist" = "Ubuntu" ]; then
+		add_extra debun_extra_debian
+	elif [ "$dist" = "CentOS" ]; then
+		add_extra debun_extra_redhat
+	elif [ "$dist" = "RedHatEnterpriseServer" ]; then
+		add_extra debun_extra_redhat
+	elif [ "$dist" = "RHEL" ]; then
+		add_extra debun_extra_redhat
+	elif [ "$dist" = "OracleServer" ]; then
+		add_extra debun_extra_redhat
+	elif [ "$dist" = "EnterpriseEnterpriseServer" ]; then
+		add_extra debun_extra_redhat
+	elif [ "$dist" = "SUSE LINUX" ]; then
+		add_extra debun_extra_suse
+	elif [ "$dist" = "Slackware" ]; then
+		add_extra debun_extra_slackware
+	else
+		echo "Unknown Distro, perhaps unsupported"
+	fi
+
+	add_extra debun_extra_genlinux
+}
+
+debun_extra_gensolaris () {
+	sysdef >${tmpdir}/sys.sysdef
+	kstat >${tmpdir}/sys.kstat
+	cp /etc/release ${tmpdir}/sys.release
+	if is_available showrev ; then
+		showrev >${tmpdir}/sys.showrev
+	fi
+}
+
+### Here comes solaris specific parts
+debun_solaris () {
+	add_extra debun_extra_gensolaris
+
+	pkginfo | grep -i syslog > ${tmpdir}/pkg.packages
+
+	if is_available svcadm ; then
+		service_stop="svcadm disable syslog-ng"
+		service_start="svcadm enable syslog-ng"
+		service_status="svcs syslog-ng"
+	fi
+}
+
+debun_extra_freebsd() {
+	:
+}
+
+debun_freebsd() {
+	add_extra debun_extra_freebsd
+}
+
+debun_extra_hpux () {
+	sysdef >${tmpdir}/sys.sysdef
+	swlist >${tmpdir}/sys.swlist
+}
+
+debun_hpux () {
+	add_extra debun_extra_hpux
+}
+
+debun_extra_aix () {
+	alog -o -t console >${tmpdir}/sys.console-log
+}
+
+debun_aix () {
+	add_extra debun_extra_aix
+
+	echo "Check package files integrity"
+	rpm -V syslog-ng-premium-edition && echo "syslog-ng-premium-edition: Package files are intact"
+	rpm -V syslog-ng-premium-edition-client && echo "syslog-ng-premium-edition-client: Package files are intact"
+	rpm -V syslog-ng-premium-edition-compact && echo "syslog-ng-premium-edition-compact: Package files are intact"
+	echo "list syslog-related packages"
+	rpm -qa |grep syslog > ${tmpdir}/rpm.packages
+}
+
+detect_env_linux () {
+	if is_available lsb_release ; then
+		lsb_release -a | tee ${tmpdir}/lsb.all
+		dist=`lsb_release -si`
+	elif [ -r /etc/debian_version ]; then
+		dist="Debian"
+	elif [ -r /etc/redhat-release ]; then
+		dist="RHEL"
+	elif [ -r /etc/slackware-version ]; then
+		dist="Slackware"
+	elif [ -r /etc/SuSE-release ]; then
+		dist="SUSE LINUX"
+	fi
+}
+
+detect_env () {
+	###
+	### Detecting syslog-ng ver: ose or pe
```

Comment:
Actually it doesn't detect if it's ose or pe rather the install directory structure if it follows fhs, like
a) /etc, /usr/sbin, /var/whatever
b) /usr/local/ whaterver
c) /opt/syslog-ng/whatever
ose installations are a) if they come from OS package as part of the system, and managed by distro's package management
local installations from source usually follows b)
and PE installations follows c)